lib/mwUtil.js

```diff
@@ -102,5 +102,58 @@ mwUtil.getLimit = function(hyper, req) {
     return hyper.config.default_page_size;
 };
 
+/**
+ * Applies access checks to the item. Throws appropriate HTTPError
+ * if access is restricted, no-op otherwise.
+ *
+ * Item might be either a revision or a restriction.
+ *
+ * @param {Object} item the item to check restriction on
+ * @param {string} [targetTid] the item tid.
+ *                             Used in case item is a restriction to indicate the content tid.
+ */
+mwUtil.applyAccessChecks = function(item, targetTid) {
+    // Page was deleted - new version of tracking
+    if (item.page_deleted) {
+        var deletionTime = uuid.fromString(item.page_deleted).getDate();
+        var revisionTime = uuid.fromString(targetTid || item.tid).getDate();
```

Comment:
I'm not completely sure that this is actually getting the timestamp corresponding to the original revision, rather than the time at which we saved the revision metadata / restrictions.